core/src/main/scala/com/databricks/labs/remorph/generators/sql/ExpressionGenerator.scala

```diff
@@ -11,63 +11,64 @@ class ExpressionGenerator extends BaseSQLGenerator[ir.Expression] with Transform
+      case r: ir.RLike => rlike(r)
+      case _: ir.Bitwise => bitwise(expr)
+      case _: ir.Arithmetic => arithmetic(expr)
+      case b: ir.Between => between(b)
+      case _: ir.Predicate => predicate(expr)
+      case l: ir.Literal => literal(l)
+      case a: ir.ArrayExpr => arrayExpr(a)
+      case m: ir.MapExpr => mapExpr(m)
+      case s: ir.StructExpr => structExpr(s)
+      case i: ir.IsNull => isNull(i)
+      case i: ir.IsNotNull => isNotNull(i)
       case ir.UnresolvedAttribute(name, _, _, _, _, _, _) => lift(OkResult(name))
-      case d: ir.Dot => dot(ctx, d)
-      case i: ir.Id => id(ctx, i)
-      case o: ir.ObjectReference => objectReference(ctx, o)
-      case a: ir.Alias => alias(ctx, a)
-      case d: ir.Distinct => distinct(ctx, d)
-      case s: ir.Star => star(ctx, s)
-      case c: ir.Cast => cast(ctx, c)
-      case t: ir.TryCast => tryCast(ctx, t)
-      case col: ir.Column => column(ctx, col)
+      case d: ir.Dot => dot(d)
+      case i: ir.Id => id(i)
+      case o: ir.ObjectReference => objectReference(o)
+      case a: ir.Alias => alias(a)
+      case d: ir.Distinct => distinct(d)
+      case s: ir.Star => star(s)
+      case c: ir.Cast => cast(c)
+      case t: ir.TryCast => tryCast(t)
+      case col: ir.Column => column(col)
       case _: ir.DeleteAction => code"DELETE"
-      case ia: ir.InsertAction => insertAction(ctx, ia)
-      case ua: ir.UpdateAction => updateAction(ctx, ua)
-      case a: ir.Assign => assign(ctx, a)
-      case opts: ir.Options => options(ctx, opts)
-      case i: ir.KnownInterval => interval(ctx, i)
-      case s: ir.ScalarSubquery => scalarSubquery(ctx, s)
-      case c: ir.Case => caseWhen(ctx, c)
-      case w: ir.Window => window(ctx, w)
-      case o: ir.SortOrder => sortOrder(ctx, o)
-      case ir.Exists(subquery) => code"EXISTS (${ctx.logical.generate(ctx, subquery)})"
-      case a: ir.ArrayAccess => arrayAccess(ctx, a)
-      case j: ir.JsonAccess => jsonAccess(ctx, j)
-      case l: ir.LambdaFunction => lambdaFunction(ctx, l)
-      case v: ir.Variable => variable(ctx, v)
-      case s: ir.SchemaReference => schemaReference(ctx, s)
-      case r: ir.RegExpExtract => regexpExtract(ctx, r)
-      case t: ir.TimestampDiff => timestampDiff(ctx, t)
-      case t: ir.TimestampAdd => timestampAdd(ctx, t)
-      case e: ir.Extract => extract(ctx, e)
-      case c: ir.Concat => concat(ctx, c)
-      case i: ir.In => in(ctx, i)
+      case ia: ir.InsertAction => insertAction(ia)
+      case ua: ir.UpdateAction => updateAction(ua)
+      case a: ir.Assign => assign(a)
+      case opts: ir.Options => options(opts)
+      case i: ir.KnownInterval => interval(i)
+      case s: ir.ScalarSubquery => scalarSubquery(s)
+      case c: ir.Case => caseWhen(c)
+      case w: ir.Window => window(w)
+      case o: ir.SortOrder => sortOrder(o)
+      case ir.Exists(subquery) =>
+        withGenCtx.flatMap(ctx => code"EXISTS (${ctx.logical.generate(subquery)})")
```

Comment:
ok, if this keeps working - i'm fine. btw, what do you think about this sugar:
```suggestion
      case ir.Exists(subquery) => withGeneratorContext { ctx =>
        code"EXISTS (${ctx.logical.generate(subquery)})"
      }
```

`withGeneratorContext` to be on `CodeGenerator` trait

Reply:
You're right, that's more convenient. Done.